src/plugins/data/public/utils/table_inspector_view/components/data_view.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+
+import { DataTableFormat } from './data_table';
+import { IUiSettingsClient } from '../../../../../../core/public';
+import {
+  Datatable,
+  DatatableColumn,
+} from '../../../../../expressions/common/expression_types/specs';
+import { InspectorViewProps, Adapters } from '../../../../../inspector/public';
+import { UiActionsStart } from '../../../../../ui_actions/public';
+import { FieldFormatsStart } from '../../../field_formats';
+import { TablesAdapter } from '../../../../../expressions/common';
+
+interface DataViewComponentState {
+  datatable: Datatable;
+  adapters: Adapters;
+}
+
+interface DataViewComponentProps extends InspectorViewProps {
+  uiSettings: IUiSettingsClient;
+  uiActions: UiActionsStart;
+  fieldFormats: FieldFormatsStart;
+  isFilterable: (column: DatatableColumn) => boolean;
+}
+
+class DataViewComponent extends Component<DataViewComponentProps, DataViewComponentState> {
+  static propTypes = {
+    adapters: PropTypes.object.isRequired,
+    title: PropTypes.string.isRequired,
+    uiSettings: PropTypes.object,
+    uiActions: PropTypes.object.isRequired,
+    fieldFormats: PropTypes.object.isRequired,
+    isFilterable: PropTypes.func.isRequired,
+  };
+
+  state = {} as DataViewComponentState;
+  _isMounted = false;
+
+  static getDerivedStateFromProps(
+    nextProps: Readonly<DataViewComponentProps>,
+    state: DataViewComponentState
+  ) {
+    if (state && nextProps.adapters === state.adapters) {
```

Comment:
Is this condition ever actually met with the strict equality check between two objects? Or is `nextProps.adapters` always going to be a new reference even if the values haven't changed? TBH I'm not sure what React does here.

Reply:
not sure, leftover from the old component